src/qtt/instrument_drivers/virtualAwg/virtual_awg.py

```diff
@@ -479,15 +495,17 @@ def sequence_gates(self, sequences, do_upload=True):
                 if awg_number != number:
                     continue
 
-                sequence_data = Sequencer.get_data(sequence, sampling_rate)
+                waveforms = [Sequencer.get_data(waveform, sampling_rate) for waveform in sequence]
+                sequence_data = np.sum(waveforms, 0)
                 sequence_data = sequence_data[:-1]
                 if not marker_number:
-                    awg_to_gate = self._settings.parameters['awg_to_{}'.format(gate_name)].get()
+                    awg_to_gate = self._settings.parameters[f'awg_to_{gate_name}'].get()
                     scaling_ratio = 1 / (awg_to_gate * gain_factor)
                     settings_data[number][gate_name] = {'scaling_ratio': scaling_ratio}
                     sequence_data *= scaling_ratio
 
-                sequence_names.append('{}_{}'.format(gate_name, sequence['name']))
+                sequence_name = sequence['name']
```

Comment:
```suggestion
                sequence_name = sequence'[0]['name']
```
`sequence` is a list now. I picked the first element to generate a name. An alternative would be
`'_'.join([s['name'] for s in sequence])`